### PR TITLE
Logging cogs

### PR DESCRIPTION
**Pull request by 1mn**

### CODE REVIEW

File: bot/cogs/clickup.py [hunk 1]
```diff
@@ -94,15 +96,20 @@ def __init__(self, bot: AutoShardedBot):
...
+                      f"error: {result['err']}")
             embed.description = f"`{result['ECODE']}`: {result['err']}"
             embed.colour = Colour.red()
 
         else:
             tasks = result["tasks"]
 
             if not tasks:
-                embed.colour = Colour.red()
+                log.warning("{ctx.author} requested a list of ClickUp tasks, but no ClickUp tasks were found!")
```
Comment (JoeBanks13) [inline bot/cogs/clickup.py]: Incorrect level, having an empty task list is not an error. If a task list is actually empty then it's not an error or warning.

File: bot/cogs/clickup.py [hunk 2]
```diff
@@ -112,6 +119,8 @@ def __init__(self, bot: AutoShardedBot):
                     status = f"{task['status']['status'].title()}"
 
                     lines.append(f"{id_fragment} ({status})\n\u00BB {task['name']}")
+
+                log.warning(f"{ctx.author} requested a list of ClickUp tasks. Returning list.")
```
Comment (JoeBanks13) [inline bot/cogs/clickup.py]: Incorrect level, this should be an info/debug log, no warnings or errors have occured to reach this code.

File: bot/cogs/verification.py [hunk 3]
```diff
@@ -21,18 +24,24 @@ def __init__(self, bot: AutoShardedBot):
...
 
         if ctx.channel.id == VERIFICATION_CHANNEL:  # We're in the verification channel
             for role in ctx.author.roles:
                 if role.id == VERIFIED_ROLE:
+                    log.warning(f"{ctx.author} posted '{ctx.message.content}' "
+                                "in the verification channel, but is already verified")
                     return  # They're already verified
 
+            log.debug(f"{ctx.author} posted '{ctx.message.content}' in the verification "
+                      "channel, we is providing instructions how to verify")
```
Comment (JoeBanks13) [inline bot/cogs/verification.py]: Is that a typo, should it be `we are providing...`

File: bot/cogs/clickup.py (hunk 1 above)
Comment (1mn) [reply bot/cogs/clickup.py]: yep, agreed

**Comment by 1mn** [comment]
Tested all commands thoroughly, and fixed a number of bugs and situations that occurred during testing.

Please note that this PR has a couple of minor fixes that would normally be considered out of scope, and that is intentional.

File: bot/__init__.py [hunk 4]
```diff
@@ -2,28 +2,65 @@
...
+
+
+def monkeypatch_trace(self, msg, *args, **kwargs):
+    """
+    Log 'msg % args' with severity 'TRACE'.
+
+    To pass exception information, use the keyword argument exc_info with
+    a true value, e.g.
+
+    logger.trace("Houston, we have a %s", "interesting problem", exc_info=1)
```
Comment (AnonGuy) [inline bot/__init__.py]: Can this be changed to `Houston, we have an...` :D

File: bot/__init__.py [hunk 5]
```diff
@@ -69,10 +106,35 @@ def _get_word(self) -> str:
...
+    python_parse_conditions = (
+        current == "("
+        and next
+        and next != ")"
+        and syntax_valid
+    )
+
+    if python_parse_conditions:
+        log.debug(f"A python-style command was used. Attempting to parse. Buffer is {self.buffer}. "
+                  f"A step-by-step can be found in the trace log.")
```
Comment (AnonGuy) [inline bot/__init__.py]: Does this string need to be an f-string, or was it left for consistency?

File: bot/cogs/bot.py [hunk 6]
```diff
@@ -85,16 +89,20 @@ def codeblock_stripping(self, msg: str):
         if msg.count("\n") >= 3:
             # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found
             if re.search("```(python|py)\n((?:.*\n*)+)```", msg, re.IGNORECASE):
+                log.trace(f"Someone wrote a message that was already a "
```
Comment (AnonGuy) [inline bot/cogs/bot.py]: Why is this an f-string?

File: bot/cogs/bot.py [hunk 7]
```diff
@@ -103,41 +111,50 @@ def codeblock_stripping(self, msg: str):
...
+                                 "I noticed you were trying to paste code into this channel.\n\n"
+                                 f"{codeblock_tag['tag_content']}")
+
+                        howto_embed = Embed(description=howto)
+                        await msg.channel.send(embed=howto_embed)
+                        self.channel_cooldowns[msg.channel.id] = time.time()
+
+                except SyntaxError:
+                    log.trace(f"{msg.author} posted in a help channel, and when we tried to parse it as Python code, "
+                              f"ast.parse raised a SyntaxError. This probably just means it wasn't Python code. "
```
Comment (AnonGuy) [inline bot/cogs/bot.py]: Does this middle line need to be an f-string?

File: bot/cogs/deployment.py [hunk 8]
```diff
@@ -66,9 +75,10 @@ def __init__(self, bot: AutoShardedBot):
                 name=key, value=value, inline=True
             )
 
+        log.debug(f"Uptimes retrieved and parsed, returning data.")
```
Comment (AnonGuy) [inline bot/cogs/deployment.py]: Does this need to be an f-string?

File: bot/cogs/tags.py [hunk 9]
```diff
@@ -240,26 +271,31 @@ def _command_on_cooldown(tag_name) -> bool:
...
+            log.debug(f"{ctx.author} successfully deleted the tag called '{tag_name}'")
             embed.colour = Colour.blurple()
             embed.title = tag_name
             embed.description = f"Tag successfully removed: {tag_name}."
 
+        elif tag_data.get("success") is False:
+            log.debug(f"{ctx.author} tried to delete a tag called '{tag_name}', but the tag does not exist.")
+            embed.colour = Colour.red()
+            embed.title = tag_name
+            embed.description = f"Tag doesn't appear to exist."
```
Comment (AnonGuy) [inline bot/cogs/tags.py]: Any reason this is an f-string?

File: bot/decorators.py [hunk 10]
```diff
@@ -1,31 +1,49 @@
...
+                  f"the '{ctx.command.name}' command, so the request is rejected.")
         return False
     return commands.check(predicate)
 
 
 def without_role(*role_ids: int):
     async def predicate(ctx: Context):
         if not ctx.guild:  # Return False in a DM
+            log.debug(f"{ctx.author} tried to use the '{ctx.command.name}' command from a DM. "
+                      f"This command is restricted by the without_role decorator. Rejecting request.")
```
Comment (AnonGuy) [inline bot/decorators.py]: Any reason this is an f-string?

**Pull request merged**